### DISCUSSION TITLE
Running FastAPI-Users behind a reverse proxy

### DISCUSSION THREAD

**Discussion by Iskarioti** [fastapi-users/fastapi-users#1362] [General]
I am utilizing FastAPI-Users within an Nginx reverse proxy setup where the root is set to "/api-service". Nevertheless, the application constructs the callback/redirect URI without considering the reverse proxy root. What steps should I take to address this problem?

**Comment by hgalytoby** [comment]
Can you provide the code for FastAPI-Users and the Nginx settings?

**Reply by Iskarioti** [reply]
Below is the nginx configuration for the proxy

``` 
location /api-service {
        proxy_http_version 1.1;
        proxy_set_header Host $host;
        proxy_set_header Connection "upgrade";
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

        proxy_pass http://localhost:8090;

        rewrite /svcs/(.*)$ /$1 break;
        proxy_redirect off;
    }

    location / {
        root /home/web-ui/build;
        index index.html index.htm;
        try_files $uri $uri/ /index.html =404;
    }
```
Then for FastAPI-Users, I am just using the basic setup as outlined in the docs.

**Comment by frankie567** [comment]
I don't think it's specifically related to FastAPI Users. When running behind a proxy, there are a few things to take into account.

I'll suggest you to take a look at Uvicorn documentation: https://www.uvicorn.org/deployment/#running-behind-nginx

Regarding proxy root, I think you should set the `--root-path` setting on Uvicorn: https://www.uvicorn.org/settings/#http

**Comment by hgalytoby** [comment]
Indeed, I don't think this has anything to do with FastAPI Users in particular either.

I have no problems using FastAPI Users with Nginx.

I don't have a deep understanding of how Nginx and Uvciron work and how they are set up, so I don't know everything about your Nginx setup.

I don't know much about Nginx usage, and I don't know if you know much about Nginx.

But when I use Nginx, the front-end and back-end locations are separate.

I see that your location is written the same and I don't know if there is any problem.

You can try it first, create a simple API and setup only the backend in Nginx.




Or you can try to see where to modify FastAPI.

For example your API is 

```
/api-service/fastapi/jwt/login
/api-service/fastapi/register
/api-service/fastapi/any-api
```


```
location /api-service/fastapi {
    ...
    proxy_pass http://localhost:8090;
    ...
}

location /api-service {
    ...
    index index.html index.htm;
    ...
}
```
